src/time_bucket.c

```diff
@@ -487,3 +492,32 @@ ts_time_bucket_ng_timezone(PG_FUNCTION_ARGS)
 
 	PG_RETURN_DATUM(DirectFunctionCall2(timestamp_zone, tzname, TimestampGetDatum(result)));
 }
+
+TS_FUNCTION_INFO_V1(ts_time_bucket_ng_timezone_origin);
+TSDLLEXPORT Datum
+ts_time_bucket_ng_timezone_origin(PG_FUNCTION_ARGS)
+{
+	Timestamp result;
+	Datum timestamp, origin;
+	Datum interval = PG_GETARG_DATUM(0);
+	Datum timestamptz = PG_GETARG_DATUM(1);
+	Datum origintz = PG_GETARG_DATUM(2);
+	Datum tzname = PG_GETARG_DATUM(3);
```

Comment:
Shouldn't these be of the corresponding datatype instead of `Datum`?

Reply:
e.g. `PG_GETARG_TIMESTAMPTZ`

Reply:
Here there is no point converting them to TImestampTz and back to Datum. We never access these arguments as TimestampTz. We only pass them to other functions, which accept Datum when using C API.